src/evosim/matchers.py

```diff
@@ -0,0 +1,100 @@
+    applied.
+
+    Example:
+
+        >>> import evosim
+        >>> cps = evosim.supply.random_charging_points(5, seed=1)
+        >>> evs = evosim.electric_vehicles.random_electric_vehicles(10, seed=1)
+        >>> evosim.matchers.charging_point_availability(None, cps.loc[0])
+        True
+        >>> evosim.matchers.charging_point_availability(None, cps)
+        0    True
+        1    True
+        2    True
+        3    True
+        4    True
+        dtype: bool
+
+    This matcher must still conform to the same interface as all other matchers, so that
+    it can be used with others via the factory.
+    """
+    return charging_point.occupancy < charging_point.capacity
+
+
+def socket_compatibility(vehicle, charging_point) -> bool:
+    """True if vehicle and charging point are compatible."""
+    return vehicle.socket == charging_point.socket
+
+
+def distance(vehicle, charging_point, max_distance: float = 1) -> bool:
+    """Maximum distance between current vehicle location and charging point."""
+    from evosim.objectives import distance
+
+    return distance(vehicle, charging_point) < max_distance
+
+
+def distance_from_destination(vehicle, charging_point, max_distance: float = 1) -> bool:
+    """Maximum distance between vehicle destination and charging point."""
+    from evosim.objectives import distance
+
+    return (
+        distance(
+            vehicle[["dest_lat", "dest_long"]].rename(
+                columns=dict(dest_lat="latitude", dest_long="longitude")
+            ),
+            charging_point,
+        )
+        < max_distance
+    )
+
+
+def charger_compatibility(vehicle, charging_point) -> bool:
+    """True if vehicle and charging point are compatible."""
+    return vehicle.charger == charging_point.charger
+
+
+def single_factory(settings: Union[Text, Mapping]) -> Callable[[Any, Any], bool]:
+    """Transforms a string into a single matcher."""
+    from evosim import matchers
+    from functools import partial
+
+    if isinstance(settings, Text):
+        return getattr(matchers, settings)
+    if "name" not in settings:
+        raise ValueError("If a mapping, settings should have a `name` key-value pair.")
+
+    parameters = dict(**settings)
+    function = getattr(matchers, parameters.pop("name"))
+    if len(parameters) == 0:
+        return function
+    return partial(function, **parameters)
+
+
+def factory(
+    settings: Union[Sequence[Union[Text, Mapping]], Union[Text, Mapping]]
+) -> Callable[[Any, Any], bool]:
+    """Transforms a sequence of strings into a sequence of matchers."""
+    if isinstance(settings, (Text, Mapping)):
+        return single_factory(settings)
+
+    functions = [single_factory(setting) for setting in settings]
+
+    def match(vehicle, charging_point) -> bool:
+        result = functions[0](vehicle, charging_point)
+        for function in functions[1:]:
+            result &= function(vehicle, charging_point)
+        return result
```

Comment:
```suggestion
        return all(function(vehicle, charging_point) for function in functions)
```
I think this is easier to read and short-circuits